testsrc/org/mozilla/javascript/tests/PullRequest857Test.java

```diff
@@ -0,0 +1,53 @@
+package org.mozilla.javascript.tests;
+
+import static org.junit.Assert.assertNotNull;
+import java.lang.reflect.Method;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mozilla.javascript.Context;
+import org.mozilla.javascript.NativeJavaObject;
+import org.mozilla.javascript.Scriptable;
+
+
+public class PullRequest857Test{
+	Context cx=null;
+	Scriptable topScope=null;
+	@Before
+	public void enterContext(){
+		cx=Context.enter();
+		topScope = cx.initStandardObjects();
+	}
+	@After
+	public void exitContext() {
+		cx.exit();
+	}
+	
+	interface C {
+		int methodInC(String str);
+	}
+	interface B extends C {
+	}
+	public abstract class A implements B{
+	}
+	@Test
+	public void testOverrideMethodInMultiLayerInterface() {

```

Comment:
If I do this, won't it compile without the catch?
```java
	public void testOverrideMethodInMultiLayerInterface() throws NoSuchMethodException {
```

Reply:
If you do that then it'll compile without the catch and the test will fail if it actually throws the exception. The way it is now, the test will still pass even if it throws the exception.

Unless the test is explicitly checking to see if the exception is thrown, then adding the exception to the throws clause is fine.